projects/igniteui-angular/src/lib/tabs/tab-item.component.ts

```diff
@@ -289,15 +313,68 @@ export class IgxTabItemComponent extends IgxTabItemBase implements AfterViewInit
         this.tabindex = newValue ? 0 : -1;
     }
 
-    private onKeyDown(isLeftArrow: boolean, index = null): void {
+    private onKeyDown(key: string): void {
```

Comment:
Remove the different HostListeners for keydown event and add one for all keys here.